tiledb/sm/array_schema/dimension.cc

```diff
@@ -1331,6 +1395,45 @@ ByteVecValue Dimension::map_from_uint64(
   return ret;
 }
 
+template <>
+ByteVecValue Dimension::map_from_uint64<double>(
+    const Dimension* dim, uint64_t value, int bits, uint64_t max_bucket_val) {
+  assert(dim != nullptr);
+  assert(!dim->domain().empty());
+  (void)bits;  // Not needed here
+
+  ByteVecValue ret(sizeof(double));
+
+  auto dom_start = *(const double*)dim->domain().start() / 2;
+  auto dom_end = *(const double*)dim->domain().end() / 2;
+  double dom_range = dom_end - dom_start;
+
+  // Essentially take the largest value in the bucket
```

Comment:
Do you mean minimum value of the next bucket?

Reply:
@stavrospapadopoulos no, we use this method to get the splitting value and it needs to be the largest value of this bucket for the rest of the operation to work.